### PR TITLE
refactor(Action): add 'T' type param to Action.

### PR DESCRIPTION
**Pull request by saneyuki**

### CODE REVIEW

File: src/Scheduler.ts [hunk 1]
```diff
@@ -6,6 +6,6 @@ export interface Scheduler {
   schedule<T>(work: (state?: T) => Subscription | void, delay?: number, state?: T): Subscription;
   flush(): void;
   active: boolean;
-  actions: Action[];
+  actions: Action<any>[]; // XXX: use `any` to remove type param `T` from `Scheduler`.
```
Comment (saneyuki) [inline src/Scheduler.ts]: I chose this approach to keep compatibility without a breaking change.

Should we add `T` to `Scheduler` for more strongly typing?

**Comment by blesh** [comment]
I'm unsure about this change. What is the goal of the PR? What issue is it fixing?

**Comment by david-driscoll** [comment]
Looks like to add type checking for the scheduler internals, so that when `state` is destructured as part of the scheduler it gets those string types too.

**Comment by blesh** [comment]
Merged with e66b2d8.. Thanks @trxcllnt!

**Comment by blesh** [comment]
lol oops wrong window.

**Comment by saneyuki** [comment]
@blesh 

> I'm unsure about this change. What is the goal of the PR? What issue is it fixing?

The goal of this PR is just what @david-driscoll says. I intend to make internal types of `Action` strongly. In the scope of this PR, I don't think to introduce a breaking change, to keep just as a polishing up.

File: src/Scheduler.ts (hunk 1 above)
Comment (saneyuki) [inline src/Scheduler.ts]: > Should we add T to Scheduler for more strongly typing?

This is a separated issue (Sorry!). [In this PR, I don't discuss about it](https://github.com/ReactiveX/rxjs/pull/1522#issuecomment-202701079)

**Comment by blesh** [comment]
I think this change is fine. The usefulness is dubious I suppose, but otherwise LGTM. :shipit:

**Comment by kwonoj** [comment]
Merged with 5b8cb3e , thanks @saneyuki

**Pull request closed without merging**